yeat/workflows/snakefiles/Pacbio

```diff
@@ -74,3 +74,37 @@ rule canu:
         canu {params.readtype_flag} {input[0]} maxThreads={threads} -p {wildcards.sample} -d {params.outdir} {params.extra_args} useGrid=false
         ln -s {wildcards.sample}.contigs.fasta {output.contigs}
         """
+
+
+rule hifiasm:
+    output:
+        contigs="analysis/{sample}/{label}/hifiasm/{sample}_contigs.fasta"
+    input:
+        "seq/input/{sample}/combined-reads.fq.gz"
+    threads: 128
+    params:
+        prefix=lambda wildcards: f"analysis/{wildcards.sample}/{wildcards.label}/hifiasm/{wildcards.sample}",
+        extra_args=lambda wildcards: config["extra_args"][wildcards.label]
+    shell:
+        """
+        hifiasm -o {params.prefix} -t {threads} {params.extra_args} {input[0]}
+        awk '/^S/{{print ">"$2;print $3}}' {params.prefix}.bp.p_ctg.gfa > {params.prefix}.bp.p_ctg.fa
```

Comment:
This awk one-liner is pretty gnarly. Given that the shell recipes are only 3 lines long, maybe you could consider changing it from `shell` code to Python (`run`) code sprinkled with `shell()` function calls as necessary?

Reply:
I like your idea. 

`awk` does make this command a bit insane. I took this code from the `hifiasm` documentation. https://hifiasm.readthedocs.io/en/latest/faq.html#how-do-i-get-contigs-in-fasta

Hifiasm and Hifiasm-meta only output gfa files. To get the contigs, you have to run the awk command to retrieve the sequences. 

If we want, we can merge this PR now since all tests are passing (rerunning the tests will take a while) and I can add this suggestion in my next PR #50 , which is almost ready for review as well.

In the meantime, I'll be working on this suggestion.

Reply:
That works for me!